ingestion-edge/ingestion_edge/publish.py

```diff
@@ -131,34 +136,38 @@ def submit(q: SQLiteAckQueue, topic: str, **kwargs) -> Tuple[str, int]:
         if value is not None
     }
     try:
-        _publish(topic, data, attrs)
+        await _publish(topic, data, attrs)
     except TRANSIENT_ERRORS:
         # transient api call failure, write to queue
         q.put((topic, data, attrs))
-    return "", 200
+    return response.raw(b"")
```

Comment:
I probably should have noticed this before, but should this be [204 No Content](https://http.cat/204) instead?

Reply:
I would like to return 204, but [the spec says return 200](https://github.com/mozilla/gcp-ingestion/blob/master/docs/edge.md#postput-response-codes)

That said, I think the above `try` clause needs an `except: return response.raw(b"", 400)`, or something because a non-transient request failure indicates a bad request that should not be retried.

Reply:
perhaps [422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) is appropriate here?

Reply:
422 is part of [RFC4918](https://tools.ietf.org/html/rfc4918#section-11.2), which is for WebDAV. I try and avoid WebDAV. I think just 400 would be fine, since there aren't really multiple failure cases here.

Reply:
eh, spec says 400, can be updated to 422 later if needed